rasa_core/utils.py

```diff
@@ -396,6 +399,23 @@ def zip_folder(folder):
     return shutil.make_archive(zipped_path.name, str("zip"), folder)
 
 
+def unarchive(byte_array: bytes, directory: Text):
+    """Tries to unpack a byte array interpreting it as an archive.
+
+    Tries to use tar first to unpack, if that fails, zip will be used."""
+
+    try:
+        tar = tarfile.open(fileobj=IOReader(byte_array))
+        tar.extractall(directory)
+        tar.close()
+        return directory
+    except tarfile.TarError:
+        zip_ref = zipfile.ZipFile(IOReader(byte_array))
```

Comment:
reuse IOReader

Reply:
no that is not a good idea, because it is unclear how many bytes the tarfile already read.

Reply:
oh. makes sense :-D